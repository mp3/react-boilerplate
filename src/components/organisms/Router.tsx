import { createBrowserHistory } from 'history'
import * as React from 'react'
import { Provider } from 'react-redux'
import { ConnectedRouter } from 'react-router-redux'
import createStore from 'store/createStore'

const history = createBrowserHistory()
export const store = createStore(history)

interface Props {
  children: JSX.Element
}

export default (props: Props) => {
  const { children } = props
  return (
    <Provider store={store}>
      <ConnectedRouter history={history}>{children}</ConnectedRouter>
    </Provider>
  )
}
