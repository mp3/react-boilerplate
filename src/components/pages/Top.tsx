import Top from 'components/templates/Top'
import * as React from 'react'

export default (props: any) => {
  return <Top />
}
