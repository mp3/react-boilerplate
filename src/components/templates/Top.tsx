import Counter from 'components/organisms/counter'
import * as React from 'react'
import styled from 'styled-components'

export default (props: any) => {
  return (
    <Container>
      <Counter />
    </Container>
  )
}

const Container = styled.div`
  padding-top: 56px;
  text-align: center;
`
