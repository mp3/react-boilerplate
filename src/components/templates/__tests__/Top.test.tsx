import * as React from 'react'
import * as renderer from 'react-test-renderer'
import Top from 'components/templates/Top'
import Router from 'components/organisms/Router'

test('render <Top />', () => {
  const tree = renderer.create(
    <Router>
      <Top />
    </Router>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
