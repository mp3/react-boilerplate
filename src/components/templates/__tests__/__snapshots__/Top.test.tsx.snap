// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <Top /> 1`] = `
<div
  className="sc-bZQynM jcKNxR"
>
  <div
    className="sc-bwzfXH bRhOHE"
  >
    <div
      className="sc-htpNat hFgGyf"
    >
      <h1
        className="sc-bxivhb bOYIrQ"
      >
        React + Redux Sample
      </h1>
      <span
        className="sc-ifAKCX cicvxl"
      >
        value: 
        0
      </span>
      <div
        className="sc-EHOje iafsfZ"
      >
        <button
          className="sc-bdVaJa LDwMf"
          onClick={[Function]}
        >
          add:3
        </button>
        Â 
        <button
          className="sc-bdVaJa LDwMf"
          onClick={[Function]}
        >
          increment
        </button>
      </div>
    </div>
  </div>
</div>
`;
